File: src/main/java/net/zomis/chess/MoveEngine.java
package net.zomis.chess;

public class MoveEngine {

    private final ChessBoard board;

    public MoveEngine(ChessBoard board) {
        this.board = board;
    }

    public Iterable<ChessPosition> getAllowedMoves(ChessTile tile) {
        switch (tile.getPiece()) {
            case PAWN:
        }
        throw new UnsupportedOperationException();
    }

    public boolean moveValid(ChessTile tile, ChessPosition pos) {
        if (tile == null) {
            return false;
            // throw new NullPointerException("tile cannot be null");
        }
        switch (tile.getPiece()) {
            case PAWN:
                return true;
            default:
                throw new UnsupportedOperationException(tile.getPiece() + " cannot move yet");
        }
    }


}

File: chess-core/src/main/kotlin/net/zomis/chess/core/MoveEngine.kt
package net.zomis.chess.core

class MoveEngine(private val board: ChessBoard) {

    fun getAllowedMoves(tile: ChessTile): Iterable<ChessPosition> {
        when (tile.piece) {
            else -> throw UnsupportedOperationException()
        }
    }

    fun moveValid(tile: ChessTile?, pos: ChessPosition): Boolean {
        if (tile == null) {
            return false
            // throw new NullPointerException("tile cannot be null");
        }
        when (tile.piece) {
            Piece.PAWN -> return true
            else -> throw UnsupportedOperationException("${tile.piece} cannot move yet")
        }
    }


}
